### PR TITLE
Add basic gradle support

### PR DESCRIPTION
**Pull request by kotarak**
Judging from where "lein" and "boot" is used, this should be a first guess to add gradle support for `cider-jack-in`. This is not ready for pull, yet. I'm not sure the changes are sufficient and complete.

https://github.com/clojure-emacs/cider/issues/1373

### CODE REVIEW

File: cider.el [hunk 1]
```diff
@@ -106,6 +106,20 @@ version from the CIDER package or library.")
   :group 'cider
   :package-version '(cider . "0.9.0"))
 
+(defcustom cider-gradle-command
+  "gradle"
+  "The command used to execute Gradle."
+  :type 'string
+  :group 'cider
+  :package-version '(cider . "0.9.0"))
```
Comment (Malabarba) [inline cider.el]: This should be 0.10.0

File: cider.el [hunk 2]
```diff
@@ -162,19 +176,22 @@ Sub-match 1 must be the project path.")
   "Check if the command matching PROJECT-TYPE is present."
   (pcase project-type
     ("lein" 'cider--lein-present-p)
-    ("boot" 'cider--boot-present-p)))
+    ("boot" 'cider--boot-present-p)
+    ("gradle" 'cider--gradle-present-p)))
```
Comment (Malabarba) [inline cider.el]: This is just personal preference, but I would have added this line before the lein entry to minimise the diff. Not sure if others agree, though.

File: cider.el [hunk 3]
```diff
@@ -424,6 +444,14 @@ In case `default-directory' is non-local we assume the command is available."
       (executable-find cider-boot-command)
       (executable-find (concat cider-boot-command ".exe"))))
 
+(defun cider--gradle-present-p ()
+  "Check if `cider-gradle-command' is on the `exec-path'.
+
```
Comment (Malabarba) [inline cider.el]: I'm not a fan of empty second lines in short docstrings.

**Comment by bbatsov** [comment]
@kotarak ping :-)

**Comment by kotarak** [comment]
@bbatsov Pong!

Searching my way through elisp and emacs. I shortly tested this. It seems to work for me with the new version of the clojuresque plugin 2.0.0. (I launched to latter to the official gradle plugin site, but it doesn't work, yet. :-/)

I also opened a [required PR for clojure-mode](https://github.com/clojure-emacs/clojure-mode/pull/340).

File: cider.el (hunk 3 above)
Comment (kotarak) [inline cider.el]: I went with this existing project style.

File: cider.el (hunk 1 above)
Comment (kotarak) [inline cider.el]: Fixed.

**Comment by bbatsov** [comment]
Looks OK - just squash the commits together and update the changelog.

File: cider.el [hunk 4]
```diff
@@ -396,16 +413,20 @@ Use `cider-ps-running-nrepls-command' and `cider-ps-running-nrepl-path-regexp-li
...
-          (lein-project-exists "lein")
-          (boot-project-exists "boot"))))
+         (choices (delq nil
+                        (mapcar (lambda (candidate)
+                                  (when (file-exists-p (cdr candidate))
+                                    (car candidate)))
+                                '(("lein" . "project.clj")
+                                  ("boot" . "build.boot")
+                                  ("gradle" . "build.gradle"))))))
+    (if (< 1 (length choices))
```
Comment (bbatsov) [inline cider.el]: I'd normally flip the condition here as you usually say "if the length of choices is more than 1". I wonder is `seq.el` has predicates like `seq-one-p` and `seq-several-p`. Those would be pretty neat. :-)

**Comment by kotarak** [comment]
Messed up sqashing.

**Pull request closed without merging**